url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/spec/args/ArgListBuilderUtil.java

```diff
@@ -0,0 +1,714 @@
+                return new TypeTuple(new EnumType(lValues), null);
+            }
+            catch (ArgSpecException e)
+            {
+                throw new CommandException(e.getMessage());
+            }
+        }
+    }
+    
+    static
+    {
+        try
+        {
+            typeProcessor.addLibraryClasses(TypeCommands.class);
+        } 
+        catch (ExtensionException e)
+        {
+            throw new Error("Internal Scripty error, could not initialize the internal type system.");
+        }
+    }
+    
+    public static class Tuple<X, Y>
+    {
+        private X x;
+        private Y y;
+
+        public Tuple(X x, Y y)
+        {
+            this.x = x;
+            this.y = y;
+        }
+
+        public X getX()
+        {
+            return this.x;
+        }
+
+        public Y getY()
+        {
+            return this.y;
+        }
+    }
+    
+    private static class TypeTuple extends Tuple<ITypeSpec, Map<String, IArgMapping>>
+    {
+        private TypeTuple(ITypeSpec x, Map<String, IArgMapping> y)
+        {
+            super(x, y);
+        }
+    }
+
+    public static class ArgListTuple extends Tuple<IArgList, Map<String, IArgMapping>>
+    {
+        public ArgListTuple(IArgList x, Map<String, IArgMapping> y)
+        {
+            super(x, y);
+        }
+    }
+    
+    public static ArgListTuple buildArgList(ScriptyStdArgList aStdArgList)
+    throws ArgSpecException
+    {
+        // We will collect the mappings we encounter in this map.
+        Map<String, IArgMapping> lMappings = new HashMap<String, IArgMapping>();
+        
+        ScriptyArg[] lFixedArgs = aStdArgList.fixed();
+        FixedArg[] lFixedSpecs = new FixedArg[lFixedArgs.length];
+        
+        ScriptyArg[] lOptionalArgs = aStdArgList.optional();
+        OptionalArg[] lOptionalSpecs = new OptionalArg[lOptionalArgs.length];
+        
+        ScriptyArg[] lNamedArgs = aStdArgList.named();
+        NamedArg[] lNamedSpecs = new NamedArg[lNamedArgs.length];
+
+        int i = 0;
+        
+        for (ScriptyArg lArg : lFixedArgs)
+        {
+            String lArgName = lArg.name();
+            String lArgType = lArg.type();
+            // Default value is ignored for fixed args.
+            // Optional flag is ignored for fixed args.
+            
+            try
+            {
+                TypeTuple lTypeInfo = (TypeTuple) typeProcessor.startNonInteractive(lArgType);
+                ITypeSpec lTypeSpec = lTypeInfo.getX();
+                Map<String, IArgMapping> lTypeMappings = lTypeInfo.getY();
+                
+                lFixedSpecs[i] = new FixedArg(lTypeSpec);
+                if(lTypeMappings != null) lMappings.putAll(lTypeMappings);
+                // Offset with 1, the mappings should skip element 0 which is the name of the command.
+                lMappings.put(lArgName, new IndexedMapping(i +1));
+
+                i++;
+            } 
+            catch (ReplEngineException e)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 572, 603.